### PR TITLE
Fixes [SYNCOPE-1110], [SYNCOPE-1123]

### PR DESCRIPTION
**Pull request by mat-ale**
Hi,

this fixes [[SYNCOPE-1110]](https://issues.apache.org/jira/browse/SYNCOPE-1110) and [[SYNCOPE-1123]](https://issues.apache.org/jira/browse/SYNCOPE-1123).

Regards,
Matteo

### CODE REVIEW

File: client/enduser/src/main/java/org/apache/syncope/client/enduser/model/CustomAttribute.java [hunk 1]
```diff
@@ -26,18 +26,18 @@
...
-    private Boolean readonly;
+    private boolean readonly;
 
     private List<String> defaultValues = new ArrayList<>();
 
     public CustomAttribute() {
     }
 
-    public Boolean getReadonly() {
+    public boolean getReadonly() {
```
Comment (ilgrosso) [inline client/enduser/src/main/java/org/apache/syncope/client/enduser/model/CustomAttribute.java]: This should be `isReadonly()`

File: client/enduser/src/main/java/org/apache/syncope/client/enduser/resources/UserSelfReadResource.java [hunk 2]
```diff
@@ -137,12 +137,13 @@ public void writeData(final Attributes attributes) throws IOException {
     private void customizeAttrs(final Set<AttrTO> attrs,
             final Map<String, CustomAttribute> customForm) {
 
-        CollectionUtils.filter(attrs, new Predicate<AttrTO>() {
-
-            @Override
-            public boolean evaluate(final AttrTO attr) {
-                return customForm.containsKey(attr.getSchema());
+        Set<AttrTO> attrsToAdd = new HashSet<>();
```
Comment (ilgrosso) [inline client/enduser/src/main/java/org/apache/syncope/client/enduser/resources/UserSelfReadResource.java]: You could use `CollectionUtils.collect()` here.

File: fit/enduser-reference/src/test/resources/protractor-conf.js [hunk 3]
```diff
@@ -19,17 +19,17 @@
 exports.config = {
   directConnect: false,
   capabilities: {
-    'browserName': 'phantomjs',
+    'browserName': 'chrome',
```
Comment (ilgrosso) [inline fit/enduser-reference/src/test/resources/protractor-conf.js]: Bad: restore `phantomjs` here!

**Comment by andrea-patricelli** [comment]
@mat-ale we have almost finished. Please put a check to call  completeUserObject only if cuastomForm is not null and not empty and add user virtual attributes management in  completeUserObject.

**Pull request closed without merging**